addons/web/controllers/main.py

```diff
@@ -491,14 +491,22 @@ def _remove_inheritance_comments(self, inherited_template):
             if re.match(COMMENT_PATTERN, comment.text.strip()):
                 comment.getparent().remove(comment)
 
-    def _read_addon_file(self, file_path):
-        """Reads the content of a file given by file_path
-        Usefull to make 'self' testable
-        :param str file_path:
+    def _read_addon_file(self, path_or_url):
+        """Read the content of a file or an 'ir.attachment' record given by
+        'path_or_url'. A 'FileNotFoundError' is raised if the path doesn't match
+        any file paths on the system nor any attachment urls.
+        :param str path_or_url:
         :returns: str
         """
-        with file_open(file_path, 'rb') as fp:
-            contents = fp.read()
+        try:
+            with file_open(path_or_url, 'rb') as fp:
+                contents = fp.read()
+        except FileNotFoundError as e:
+            [attachment] = request.env['ir.attachment'].sudo().get_serve_attachment(path_or_url, extra_fields=['raw'])
```

Comment:
I don't understand the purpose of calling `get_serve_attachment` when you can just `search` on the url, leading to simpler code all around.